at_client/lib/src/decryption_service/shared_key_decryption.dart

```diff
@@ -31,10 +31,18 @@ class SharedKeyDecryption implements AtKeyDecryption {
     atClient ??= AtClientManager.getInstance().atClient;
     if (atKey.metadata != null && atKey.metadata!.pubKeyCS != null) {
       encryptedSharedKey = atKey.metadata!.sharedKeyEnc;
-      final currentAtSignPublicKey = (await atClient
-              ?.getLocalSecondary()!
-              .getEncryptionPublicKey(atClient!.getCurrentAtSign()!))
-          ?.trim();
+      String? currentAtSignPublicKey;
+      try {
+        currentAtSignPublicKey = (await atClient
+                ?.getLocalSecondary()!
+                .getEncryptionPublicKey(atClient!.getCurrentAtSign()!))
+            ?.trim();
+      } on KeyNotFoundException {
+        throw AtPublicKeyNotFoundException(
```

Comment:
@kalluriramkumar @VJag 
should we try to look up public key from server before throwing AtPublicKeyNotFoundException ?

Reply:
@murali-shris : The encryptionPublicKey is saved to localSecondary from .keys file and then it is sync'ed to remote secondary. So it is highly unlikely that key exist in remote secondary without its presence in local secondary.